### PR TITLE
Implement rocSOLVER dsyevd for MAGMA build

### PR DESCRIPTION
**Pull request by mewall**
o Pattern after cuSOLVER implementation
  o Build is controlled using BML_ROCSOLVER
    - Like BML_CUSOLVER

### CODE REVIEW

**Review by nicolasbock — APPROVED** [review]
LGTM

(I don't have the necessary GPU hardware though to test this.)

**Review by jeanlucf22 — CHANGES_REQUESTED** [review]

**Comment by jeanlucf22** [comment]
@mewall Did you verify that none of your additions conflict with the magma build on NVIDIA?

File: CMakeLists.txt [hunk 1]
```diff
@@ -454,6 +492,9 @@ if(MAGMA_FOUND)
   if(BML_CUSOLVER)
     list(APPEND LINK_LIBRARIES ${CUDA_cusolver_LIBRARY})
   endif()
+  if(BML_ROCSOLVER)
+    list(APPEND LINK_LIBRARIES ${ROCM_rocsolver_LIBRARY})
```
[the remark this answers is not in the document]
Comment (mewall) [reply CMakeLists.txt]: I just pushed a revision. Are these comments addressed?

**Review by jeanlucf22 — APPROVED** [review]

**Comment by mewall** [comment]
I built on summit using BML_CUSOLVER. There doesn't appear to be any conflict with the NVIDIA MAGMA build. However, some other tests are failing on summit. I'll create an issue.



________________________________
From: Jean-Luc Fattebert ***@***.***>
Sent: Monday, March 7, 2022 9:05:46 AM
To: lanl/bml
Cc: Wall, Michael E; Mention
Subject: [EXTERNAL] Re: [lanl/bml] Implement rocSOLVER dsyevd for MAGMA build (PR #597)


@mewall<https://github.com/mewall> Did you verify that none of your additions conflict with the magma build on NVIDIA?

**Pull request merged**